File: back-blog/src/main/java/fr/lgtd/backblog/entities/Post.java
package fr.lgtd.backblog.entities;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinTable;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;

@Entity
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String content;
    private String title;
    private String date;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "posts_comments", joinColumns = @JoinColumn(name = "post_id"), inverseJoinColumns = @JoinColumn(name = "comment_id"))
    private Set<Comment> comments = new HashSet<>();

    public Post() {
    }

    public Post(String content, String title, String date, User user) {
        this.content = content;
        this.title = title;
        this.date = date;
        this.user = user;
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getContent() {
        return this.content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDate() {
        return this.date;
    }

    public void setDate(String date) {
        this.date = date;
    }

}

File: front/app/src/app/models/post.ts
import { Comment } from './comment';
import { User } from './user';

export class Post {
  private content: string;
  private title: string;
  private date: string;
  private user: User;
  private comments: Comment[];

  constructor(content: string, title: string, date: string, user: User) {
    this.content = content;
    this.title = title;
    this.user = user;
    this.comments = [];
  }

  public addComment(comment: Comment) {
    this.comments.push(comment);
  }

  public getContent(): string {
    return this.content;
  }

  public setContent(content: string): void {
    this.content = content;
  }

  public getTitle(): string {
    return this.title;
  }

  public setTitle(title: string): void {
    this.title = title;
  }

  public getDate(): string {
    return this.date;
  }

  public setDate(date: string): void {
    this.date = date;
  }

  public getUser(): User {
    return this.user;
  }

  public setUser(user: User): void {
    this.user = user;
  }

  public getComments(): Comment[] {
    return this.comments;
  }

  public setComments(comments: Comment[]): void {
    this.comments = comments;
  }
}
